module/VuFind/tests/unit-tests/src/VuFindTest/Section/SectionServiceTest.php

```diff
@@ -0,0 +1,173 @@
+    public function testGetSectionConfiguration(
+        string $key,
+        string $configPath,
+        string $expectedExceptionClass,
+        ?string $expectedExceptionMsg = null
+    ) {
+        $this->expectException($expectedExceptionClass);
+        if ($expectedExceptionMsg) {
+            $this->expectExceptionMessage($expectedExceptionMsg);
+        }
+        $this->getSectionService()->getSectionConfig($key, $configPath);
+    }
+
+    /**
+     * Data provider for testSectionConfiguration.
+     *
+     * @return \Iterator<string, array>
+     */
+    public static function sectionConfigurationProvider(): \Iterator
+    {
+        yield 'Missing section type' => [
+            'MissingSectionType',
+            [],
+            BadConfig::class,
+            'Missing required setting: type',
+        ];
+        yield 'Navigation plugin with a missing plugin setting' => [
+            'MissingNavigationPlugin',
+            ['type' => 'navigation'],
+            BadConfig::class,
+            'Missing required setting: plugin',
+        ];
+        yield 'Nonexistent navigation plugin' => [
+            'NonexistentNavigationPlugin',
+            [
+                'type' => 'navigation',
+                'plugin' => 'nonexistentNavigationPlugin',
+            ],
+            ServiceNotFoundException::class,
+        ];
+    }
+
+    /**
+     * Test section configuration.
+     *
+     * @param string  $key                    Section key
+     * @param array   $config                 Configuration
+     * @param string  $expectedExceptionClass Expected exception class
+     * @param ?string $expectedExceptionMsg   Expected exception message
+     *
+     * @return void
+     */
+    #[\PHPUnit\Framework\Attributes\DataProvider('sectionConfigurationProvider')]
+    public function testSectionConfiguration(
+        string $key,
+        array $config,
+        string $expectedExceptionClass,
+        ?string $expectedExceptionMsg = null
+    ) {
+        $this->expectException($expectedExceptionClass);
+        if ($expectedExceptionMsg) {
+            $this->expectExceptionMessage($expectedExceptionMsg);
+        }
+        $this->getSectionService()->getSection($key, $config);
+    }
+
+    /**
+     * Test settings localization.
+     *
+     * @return void
+     */
+    public function testSettingsLocalization()
+    {
+        $config = [
+            'Account' => [
+                'label' => 'Your Account',
+                'MenuItems' => [
+                    [
+                        'label' => 'saved_items',
+                        'url' => [
+                            'en' => 'English language URL',
+                            'fi' => 'Finnish language URL',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $localizedConfig = $this->getAccountMenu($config)->getSectionConfig();
```

Comment:
Is it worth making a second check to confirm that it switches to Finnish under appropriate circumstances?

Reply:
I also added a third check for the fallback locales.